snuba/subscriptions/data.py

```diff
@@ -0,0 +1,46 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.factory import get_dataset
+from snuba.views import validate_request_content
+from snuba.query.types import Condition
+from snuba.request import Request
+from snuba.request.schema import RequestSchema
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """
+    Represents the state of a subscription.
```

Comment:
Would it be worth parsing the query and store a Query object in the subscription instead of another partial representation of the query (conditions + aggregations).
It would require more refactoring of the validation but then we would not need to re-parse it every time.

Reply:
This seems like it'd be a good idea. If we do this, are there any risks involved with the `Query` object changing over time?

Reply:
I like this general idea, but have a few questions/concerns with that approach:

1. The `Query` instance (I'm assuming we're talking about [this one](https://github.com/getsentry/snuba/blob/f6567d23c9295614c06c567eea0a12eb3c81f298/snuba/query/query.py#L49)) is already bound to a data source. That seems problematic for serialization, schema migrations, etc. [This comment from below](https://github.com/getsentry/snuba/pull/618/files#r354089337) would also apply here. Changing so that a query isn't bound to a dataset or data source (table, optionally with joins) would alleviate these concerns, but that's a pretty wide reaching change. At a high level, I think `source.execute(query)` makes a lot more sense to me than `query.execute()` with the source already bound, for what it's worth.
2. The query contains nodes that are variables (current timestamp.) I'm not sure how we'd express those. Adding a `Variable` node to the AST would enable that, but obviously opens another complexity can of worms.

Reply:
Also if we're planning on using `parse_and_run_query` then it might make sense to keep things in the same format.

Reply:
For the moment it seems like it makes sense to keep this as is, so that we don't blow out the complexity.

Reply:
ok, no, if this represent a long lived subscription then, definitively DO NOT use Query. 
That is a throw away object to execute the query that gets updated in place multiple times every time the query is executed. I thought this would represent something different. That represents one instance of a query.